### PR TITLE
docs(checkout): CHECKOUT-6253 Introducing promotion field for the API reference document for Carts and Checkout

### PR DESCRIPTION
**Pull request by bc-mackxu**
Main Jira ticket: https://jira.bigcommerce.com/browse/CHECKOUT-5069
API update ticket: 
- https://jira.bigcommerce.com/browse/CHECKOUT-6241
- https://jira.bigcommerce.com/browse/CHECKOUT-6242

Document update Ticket: https://jira.bigcommerce.com/browse/CHECKOUT-6253

@bigcommerce/checkout @bigcommerce/dev-docs

### CODE REVIEW

**Comment by slsriehl** [comment]
@bc-mackxu -- I don't see a DEVDOCS ticket associated with [CHECKOUT-5069](https://jira.bigcommerce.com/browse/CHECKOUT-5069).  Please create or link one so that we can give our attention to this PR.

**Comment by bc-mackxu** [comment]
> @bc-mackxu -- I don't see a DEVDOCS ticket associated with [CHECKOUT-5069](https://jira.bigcommerce.com/browse/CHECKOUT-5069). Please create or link one so that we can give our attention to this PR.

Thank you for checking, here is the document update ticket: https://jira.bigcommerce.com/browse/CHECKOUT-6253

**Comment by slsriehl** [comment]
Thanks @bc-mackxu.  I created related [DEVDOCS-3376](https://jira.bigcommerce.com/browse/DEVDOCS-3376) to get this PR in our queue for approval.

File: reference/carts.v3.yml [hunk 1]
```diff
@@ -141,10 +142,12 @@ paths:
             * `redirect_urls`: Create a direct link to a Cart. This can be used during the /POST request for Carts.
             * `line_items.physical_items.options`: The Cart returns an abbreviated result. Use this to return physical items product options. Can also be used in a /POST to have the extended Cart object return.
             * `line_items.digital_items.options`: The Cart returns an abbreviated result. Use this to return digital items product options.  Can also be used in a /POST to have the extended Cart object return.
+            * `promotions.banners`: Returns an list of eligible banner.
```
Comment (traci707) [inline reference/carts.v3.yml:145]: * `promotions.banners`: Returns a list of eligible banners.

File: reference/carts.v3.yml [hunk 2]
```diff
@@ -222,10 +225,12 @@ paths:
...
```
Comment (traci707) [inline reference/carts.v3.yml:228]: * `promotions.banners`: Returns a list of eligible banners.

File: reference/carts.v3.yml [hunk 3]
```diff
@@ -324,10 +329,12 @@ paths:
...
```
Comment (traci707) [inline reference/carts.v3.yml:332]: * `promotions.banners`: Returns a list of eligible banners.

File: reference/carts.v3.yml [hunk 4]
```diff
@@ -380,10 +387,12 @@ paths:
...
```
Comment (traci707) [inline reference/carts.v3.yml:390]: * `promotions.banners`: Returns a list of eligible banners.

File: reference/carts.v3.yml [hunk 5]
```diff
@@ -476,10 +485,12 @@ paths:
...
```
Comment (traci707) [inline reference/carts.v3.yml:488]: * `promotions.banners`: Returns a list of eligible banners.

File: reference/carts.v3.yml [hunk 6]
```diff
@@ -527,10 +538,12 @@ paths:
...
```
Comment (traci707) [inline reference/carts.v3.yml:541]: * `promotions.banners`: Returns a list of eligible banners.

**Review by traci707 — COMMENTED** [review]
LGTM. I just made editorial changes.

File: reference/carts.v3.yml [hunk 7]
```diff
@@ -1414,6 +1427,28 @@ definitions:
       locale:
         type: string
         description: Locale of the cart.
+      promotions:
+        type: object
+        description: This is available only when "include=promotions.banners" is presented in the URL.
```
Comment (slsriehl) [inline reference/carts.v3.yml:1432]:
```suggestion
        description: 'This is available only when "include=promotions.banners" is presented in the URL.'
```

File: reference/carts.v3.yml [hunk 8]
```diff
@@ -1414,6 +1427,28 @@ definitions:
...
+        properties:
+          banners:
+            title: Banner
+            type: object
+            properties:
+              id:
+                description: Id of the promotion.
+                type: string
+              type:
+                description: 'Type of the banner'
```
Comment (slsriehl) [inline reference/carts.v3.yml:1442]:
```suggestion
                description: Type of the banner
```

File: reference/carts.v3.yml [hunk 9]
```diff
@@ -1414,6 +1427,28 @@ definitions:
...
+                type: string
+              page:
+                description: 'Array of locations where the banner will be shown'
```
Comment (slsriehl) [inline reference/carts.v3.yml:1445]:
```suggestion
                description: An array of locations where the banner will display
```

File: reference/carts.v3.yml [hunk 10]
```diff
@@ -1414,6 +1427,28 @@ definitions:
...
+                type: array
+                items:
+                  type: string
+              text:
+                description: 'Text of the banner'
```
Comment (slsriehl) [inline reference/carts.v3.yml:1450]:
```suggestion
                description: Text of the banner
```

File: reference/checkouts.v3.yml [hunk 11]
```diff
@@ -1217,6 +1219,28 @@ definitions:
...
+        properties:
+          banners:
+            title: Banner
+            type: object
+            properties:
+              id:
+                description: Id of the promotion.
+                type: string
+              type:
+                description: 'Type of the banner'
```
Comment (slsriehl) [inline reference/checkouts.v3.yml:1234]:
```suggestion
                description: Type of the banner
```

File: reference/checkouts.v3.yml [hunk 12]
```diff
@@ -1217,6 +1219,28 @@ definitions:
...
+                type: string
+              page:
+                description: 'Array of locations where the banner will be shown'
```
Comment (slsriehl) [inline reference/checkouts.v3.yml:1237]:
```suggestion
                description: An array of the locations where the banner will display
```

File: reference/checkouts.v3.yml [hunk 13]
```diff
@@ -1217,6 +1219,28 @@ definitions:
...
+                type: array
+                items:
+                  type: string
+              text:
+                description: 'Text of the banner'
```
Comment (slsriehl) [inline reference/checkouts.v3.yml:1242]:
```suggestion
                description: Text of the banner
```

File: reference/carts.v3.yml (hunk 1 above)
Comment (traci707) [inline reference/carts.v3.yml:145]:
```suggestion
            * `promotions.banners`: Returns a list of eligible banners.
```

File: reference/carts.v3.yml (hunk 2 above)
Comment (traci707) [inline reference/carts.v3.yml:228]:
```suggestion
            * `promotions.banners`: Returns a list of eligible banners.
```

File: reference/carts.v3.yml (hunk 3 above)
Comment (traci707) [inline reference/carts.v3.yml:332]:
```suggestion
            * `promotions.banners`: Returns a list of eligible banners.
```

File: reference/carts.v3.yml (hunk 4 above)
Comment (traci707) [inline reference/carts.v3.yml:390]:
```suggestion
            * `promotions.banners`: Returns a list of eligible banners.
```

File: reference/carts.v3.yml (hunk 5 above)
Comment (traci707) [inline reference/carts.v3.yml:488]:
```suggestion
            * `promotions.banners`: Returns a list of eligible banners.
```

File: reference/carts.v3.yml (hunk 6 above)
Comment (traci707) [inline reference/carts.v3.yml:541]:
```suggestion
            * `promotions.banners`: Returns a list of eligible banners.
```

**Review by traci707 — COMMENTED** [review]
editorial changes

**Comment by bc-mackxu** [comment]
Thank you for reviewing @slsriehl @traci707. The feedback is done and pushed to GIT.

**Review by traci707 — APPROVED** [review]

**Review by bc-shawnwang — APPROVED** [review]
LGTM 👏

**Pull request merged**